libkbfs/md_util.go

```diff
@@ -264,5 +264,39 @@ func decryptMDPrivateData(ctx context.Context, config Config,
 		}
 		rmdToDecrypt.data = *privateMetadata
 	}
+
+	// Re-embed the block changes if it's needed.
+	if info := rmdToDecrypt.data.Changes.Info; info.BlockPointer != zeroPtr {
+		// We don't have a convenient way to fetch the block from here
+		// via folderBlockOps, so just go directly via the
+		// BlockCache/BlockOps.  No locking around the blocks is
+		// needed since these change blocks are read-only.
+		block, err := config.BlockCache().Get(info.BlockPointer)
+		if err != nil {
+			block = NewFileBlock()
+			if err := config.BlockOps().Get(ctx, rmdToDecrypt,
```

Comment:
Should this be `rmdWithKeys`? If we needed `rmdWithKeys` to decrypt the data, we might need the same key generation to decrypt the embedded block, right?